src/libraries/Common/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -19,11 +19,8 @@ internal static partial class ADP
     {
         // NOTE: Initializing a Task in SQL CLR requires the "UNSAFE" permission set (https://learn.microsoft.com/dotnet/framework/performance/sql-server-programming-and-host-protection-attributes)
         // Therefore we are lazily initializing these Tasks to avoid forcing customers to use the "UNSAFE" set when they are actually using no Async features
-        private static Task<bool>? _trueTask;
-        internal static Task<bool> TrueTask => _trueTask ??= Task.FromResult(true);
-
-        private static Task<bool>? _falseTask;
-        internal static Task<bool> FalseTask => _falseTask ??= Task.FromResult(false);
+        internal static Task<bool> TrueTask => field ??= Task.FromResult(true);
+        internal static Task<bool> FalseTask => field ??= Task.FromResult(false);
```

Comment:
`Task.FromResult<bool>` already returns a cached value, and the comment above is outdated.